src/ui/ui_render/ui_render_mixin.js

```diff
@@ -238,6 +244,10 @@ export function uiRenderMixin(kbnServer, server, config) {
         legacyMetadata,
       },
     });
+
+    response.header('content-security-policy', `script-src 'unsafe-eval' 'nonce-${nonce}'`);
```

Comment:
mapbox-gl requires following policy:

```

    response.header('content-security-policy', `script-src 'unsafe-eval' 'nonce-${nonce}'`);
    response.header('content-security-policy', `img-src blob:`);
    response.header('content-security-policy', `child-src blob:`);
```

from https://docs.mapbox.com/help/troubleshooting/how-to-use-mapbox-securely/#csp-directives

Reply:
I believe the resource loading with blobs is pretty common in WebGL, to more easily transfer textures to the GL rendering context. child-src is probably due to instantiation of the web-workers (which due most of the geometry processing and data-loading in mapbox-gl), not by script, but by string-blob.

Reply:
we might not need the unsafe-eval at all. Removing it seems to work (but needs a little more investigation)
```
   response.header('content-security-policy', `script-src 'nonce-${nonce}'`);
    response.header('content-security-policy', `img-src blob:`);
    response.header('content-security-policy', `child-src blob:`);
```

The docs seem a little out of date. Requirement for unsafe-eval was removed in `0.44` (https://github.com/mapbox/mapbox-gl-js/pull/5665)

As for the requirements for img-src and child-src directives, to allow blobs, that is explained here https://github.com/mapbox/mapbox-gl-js/issues/559#issuecomment-194418634

Reply:
see also https://github.com/mapbox/mapbox-gl-js/issues/6058